background.js

```diff
@@ -54,9 +55,56 @@ async function discardTabs() {
   };
 }
 
+/**
+ * closes any tabs that are discarded, inactive for the specified time and discardable by options
+ */
+async function closeTabs() {
+  const userOptions = await browser.storage.sync.get();
+  const options = {
+    ...defaultOptions,
+    ...userOptions,
+  };
+
+  const regexes = options.regexes.map(r => new RegExp(r));
+
+  const lastActiveLimit = Date.now() - (options.timeToClose);
+
+  let tabs = await browser.tabs.query({
+    discarded: true, // Only get previously discarded tabs!
+    active: false, // Don't discard the current tab  
+  });
+
+  tabs = tabs.filter(function(tab) {
+    // Only close pinned tabs if the user has enabled it
+    if (!options.pinned && tab.pinned) {
+      return false;
+    }
+
+    // Only close audible tabs if the user has enabled it
+    if (!options.audible && tab.audible) {
+      return false;
+    }
+
+    // Only close tabs that are inactive for the specified time
+    if (tab.lastAccessed > lastActiveLimit) {
+      return false;
+    }
+
+    // Ignore tabs the match the list of regular expressions
+    if (regexes.some((r) => r.test(tab.url))) {
+      return false;
+    };
+
+    return true;
+  });
+}
+
```

Comment:
Hmm - I'm not a fan of duplicating the code like this - I think we can combine the two functions.

If we modify the tab query so that it includes both discarded and undiscarded tabs, we can handle the two difference behaviours around line 54.

Reply:
I adden another commit where i tried to apply your suggestions. Am i doing this right?